server/api/cartItems.js

```diff
@@ -1,36 +1,88 @@
-const express = require('express');
+const express = require("express");
 const app = express();
-const { CartItem } = require('../db/index');
+const { CartItem } = require("../db/index");
 
-app.get('/', (req, res, next) => {
+// these two routes are mostly for clarity but do add some functionality. Especially when it comes to decremeting the quantity of a cartItem.
+
+// increment
+app.put("/:id/increment/", (req, res, next) => {
+  const { id } = req.params;
+  CartItem.findByPk(id)
+    .then(foundCartItem => {
+      return foundCartItem.update({ quantity: foundCartItem.quantity + 1 });
+    })
+    .then(updatedCartItem => {
+      return res.status(201).send(updatedCartItem);
+    })
+    .catch(err => next(err));
+});
+
+//decrement
+app.put("/:id/decrement/", (req, res, next) => {
+  const { id } = req.params;
+  CartItem.findByPk(id)
+    .then(foundCartItem => {
+      if (foundCartItem.quantity === 1) {
+        foundCartItem.destroy();
+      }
+      if (foundCartItem.quantity > 1) {
+        foundCartItem.update({ quantity: foundCartItem.quantity - 1 });
+      }
+    })
+    .then(updatedCartItem => {
+      return res.status(201).send(updatedCartItem);
+    })
+    .catch(err => next(err));
+});
+
+app.get("/", (req, res, next) => {
   CartItem.findAll()
     .then(cartItems => res.status(200).send(cartItems))
     .catch(err => next(err));
 });
 
-app.get('/:id', (req, res, next) => {
+app.get("/:id", (req, res, next) => {
   const { id } = req.params;
   CartItem.findOne({ where: id });
 });
 
-app.post('/', (req, res, next) => {
-  CartItem.create(req.body)
-    .then(cartItem => res.status(201).send(cartItem))
+//if you post an item to a cart where it already exists it increments the quantity
+app.post("/", (req, res, next) => {
+  const { itemId, cartId } = req.body;
+  CartItem.findOne({
+    where: {
+      itemId: itemId,
+      cartId: cartId
```

Comment:
we can just type itemId, cartId since we deconstructed those key names on line 51